### ISSUE TITLE
Is there an error in the calculation of cross-entropy loss?

### ISSUE THREAD

**Issue by Megumin1124** [justinkay/aldi#45]
if self.do_cls_dst:
            if self.cls_loss_type == "CE":    
                cls_dst_loss = cross_entropy(student_cls_logits, teacher_cls_probs)
In PyTorch's implementation of cross-entropy, the target already includes a softmax operation, but I noticed that you passed the target, which has already undergone the softmax operation, into the calculation. I'm a bit confused

**Comment by justinkay** [comment]
Thanks for the question, actually PyTorch supports post-softmax scores by default which is convenient for this use case, you can check out the documentation: https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html

```
Target: If containing class indices, shape ( ) (), ( N ) (N) or ( N , d 1 , d 2 , . . . , d K ) (N,d 1 ​ ,d 2 ​ ,...,d K ​ ) 
with K ≥ 1 K≥1 in the case of K-dimensional loss where each value should be between [ 0 , C ) [0,C).
 If containing class probabilities, same shape as the input and each value should be between [ 0 , 1 ] [0,1].
```

**Comment by Megumin1124** [comment]
def custom_cross_entropy_loss(y_pred, y_true):
    
     pred_prob = F.softmax(y_pred, dim=1)
     loss = -torch.sum(y_true * torch.log(pred_prob), dim=1).mean()

     return loss

y_pred = np.array([[0.4, 0.6],
                   [0.5, 0.7],
                   [0.2, 0.4]], dtype=np.float32)

y_true = np.array([[0.2, 0.6],
                   [0.1, 0.7],
                   [0.5, 0.4]], dtype=np.float32)

y_pred = torch.tensor(y_pred)
y_true = torch.tensor(y_true)

print(F.cross_entropy(y_pred,y_true))
print(custom_cross_entropy_loss(y_pred,y_true))

I tried it, and the cross-entropy implementation in Torch produces the same output as my custom function. This validates my point
